cli/cli/commands/enclave/connect/connect.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface"
+	"github.com/kurtosis-tech/kurtosis/contexts-config-store/store"
+	"github.com/kurtosis-tech/kurtosis/metrics-library/golang/lib/metrics_client"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	enclaveIdentifierArgKey = "enclave"
+	isEnclaveIdArgOptional  = false
+	isEnclaveIdArgGreedy    = false
+
+	kurtosisBackendCtxKey = "kurtosis-backend"
+	engineClientCtxKey    = "engine-client"
+
+	portMappingSeparatorForLogs = ", "
+)
+
+var EnclaveConnectCmd = &engine_consuming_kurtosis_command.EngineConsumingKurtosisCommand{
+	CommandStr:                command_str_consts.EnclaveConnectCmdStr,
+	ShortDescription:          "Connects enclave",
+	LongDescription:           "Connects the enclave with the given name",
+	KurtosisBackendContextKey: kurtosisBackendCtxKey,
+	EngineClientContextKey:    engineClientCtxKey,
+	Flags:                     nil,
+	Args: []*args.ArgConfig{
+		enclave_id_arg.NewEnclaveIdentifierArg(
+			enclaveIdentifierArgKey,
+			engineClientCtxKey,
+			isEnclaveIdArgOptional,
+			isEnclaveIdArgGreedy,
+		),
+	},
+	RunFunc: run,
+}
+
+func init() {
+}
+
+func run(
+	ctx context.Context,
+	kurtosisBackend backend_interface.KurtosisBackend,
+	engineClient kurtosis_engine_rpc_api_bindings.EngineServiceClient,
+	_ metrics_client.MetricsClient,
+	_ *flags.ParsedFlags,
+	args *args.ParsedArgs,
+) error {
+	enclaveIdentifier, err := args.GetNonGreedyArg(enclaveIdentifierArgKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting the enclave identifier arg using key '%v'", enclaveIdentifierArgKey)
+	}
+
+	currentContext, err := store.GetContextsConfigStore().GetCurrentContext()
+	if err != nil {
+		logrus.Warnf("Could not retrieve the current context. Kurtosis will assume context is local.")
+		logrus.Debugf("Error was: %v", err.Error())
+		return nil
+	}
+
+	if !store.IsRemote(currentContext) {
+		logrus.Info("Current context is local, not mapping enclave service ports")
+		return nil
+	}
+
+	logrus.Info("Connecting enclave...")
+	if err = connectAllEnclaveServices(ctx, enclaveIdentifier); err != nil {
+		return stacktrace.Propagate(err, "An error occurred connecting all enclave services")
+	}
+
+	logrus.Info("Enclave connected successfully")
+
+	return nil
+}
+
+func connectAllEnclaveServices(ctx context.Context, enclaveIdentifier string) error {
+
+	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating Kurtosis Context from local engine")
+	}
+
+	enclaveCtx, err := kurtosisCtx.GetEnclaveContext(ctx, enclaveIdentifier)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting an enclave context from enclave info for enclave '%v'", enclaveIdentifier)
+	}
+	allServicesMap := map[string]bool{}
+	serviceContexts, err := enclaveCtx.GetServiceContexts(allServicesMap)
```

Comment:
while the err check below should handle errors I guess its still possible for someone to make this function return a nil for `serviceContexts` even if there's no `err`. We should check for that here